tests/langchain/test_langchain_model_export.py

```diff
@@ -0,0 +1,248 @@
+import os
+import langchain
+import mlflow
+import pytest
+import json
+import transformers
+
+from contextlib import contextmanager
+from unittest import mock
+from langchain.prompts import PromptTemplate
+from langchain.chains import LLMChain
+from langchain.llms import OpenAI
+from langchain.llms import HuggingFacePipeline
+from langchain.llms.base import LLM
+from langchain.chains.base import Chain
+from pyspark.sql import SparkSession
+from typing import Any, List, Mapping, Optional, Dict
+
+TEST_CONTENT = "MLflow-test"
+
+
+class MockResponse:
+    def __init__(self, status_code, json_data):
+        self.status_code = status_code
+        self.content = json.dumps(json_data).encode()
+        self.headers = {"Content-Type": "application/json"}
+
+
+def _mock_text_completion_json():
+    # https://platform.openai.com/docs/api-reference/chat/create
+    return {
+        "id": "cmpl-uqkvlQyYK7bGYrRHQ0eXlWi7",
+        "object": "text_completion",
+        "created": 1589478378,
+        "model": "text-davinci-003",
+        "choices": [
+            {
+                "index": 0,
+                "text": TEST_CONTENT,
+                "finish_reason": "stop",
+            }
+        ],
+        "usage": {"prompt_tokens": 9, "completion_tokens": 12, "total_tokens": 21},
+    }
+
+
+def _mock_text_completion_response():
+    return MockResponse(200, _mock_text_completion_json())
+
+
+@contextmanager
+def _mock_request(**kwargs):
+    with mock.patch("requests.Session.request", **kwargs) as m:
+        yield m
+
+
+@contextmanager
+def _mock_async_request():
+    with _mock_request(return_value=_mock_text_completion_response()) as m:
+        yield m
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return tmp_path.joinpath("model")
+
+
+@pytest.fixture(scope="module")
+def spark():
+    with SparkSession.builder.master("local[*]").getOrCreate() as s:
+        yield s
+
+
+def create_huggingface_model(model_path):
+    architecture = "lordtt13/emo-mobilebert"
+    mlflow.transformers.save_model(
+        transformers_model={
+            "model": transformers.TFMobileBertForSequenceClassification.from_pretrained(
+                architecture
+            ),
+            "tokenizer": transformers.AutoTokenizer.from_pretrained(architecture),
+        },
+        path=model_path,
+    )
+    llm = mlflow.transformers.load_model(model_path)
+    prompt = PromptTemplate(
+        input_variables=["product"],
+        template="What is a good name for a company that makes {product}?",
+    )
+    hf_pipe = HuggingFacePipeline(pipeline=llm)
+    return LLMChain(llm=hf_pipe, prompt=prompt)
+
+
+def create_openai_llmchain():
+    os.environ[
+        "OPENAI_API_KEY"
+    ] = "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"  # no credit
```

Comment:
```suggestion
```

Reply:
This is needed for the tests to run. Do you have suggestions for a better place to specify this?

Reply:
During test, we mock requests and set `OPENAI_API_KEY=test`, right?

Reply:
btw is `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` a real API key?

Reply:
Langchain validates the existence of the `OPENAI_API_KEY` environment variable. We can pass in a fake one.

Reply:
Got it. Let's remove this.

Reply:
Sounds good. How to run the test on my laptop? I use
```
pytest 'tests/langchain/test_langchain_model_export.py::test_langchain_native_save_and_load_model'
```
which does not set the environment variables.

Reply:
You can run `export OPENAI_API_KEY=test`.